### PR TITLE
apt: Add option to ignore waiting phased updates

### PR DESCRIPTION
**Pull request by sndels**
Apt seems to almost always hold a few phased updates back, for Ubuntu 22.04 at least. This makes the update indicator less useful. The logic appears to be that [anything under 100% might be held back based on the machine ID](https://salsa.debian.org/apt-team/apt/-/commit/c5bc86d45e003905ef411146e66b414d26fb1ff8). I added the option to filter out potentially held back packages from the update count.

The implementation just parses the package name from the apt list lines that were previously tallied raw, e.g.
```
gnome-initial-setup/jammy-updates 42.0.1-1ubuntu2.1 amd64 [upgradable from: 42.0.1-1ubuntu2]
```
finds the phase % from apt-cache if it's there, e.g.
```
$ apt-cache policy grub-efi-amd64-bin
grub-efi-amd64-bin:
  Installed: 2.06-2ubuntu10
  Candidate: 2.06-2ubuntu14
  Version table:
     2.06-2ubuntu14 500 (phased 24%)
        500 http://fi.archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages
 *** 2.06-2ubuntu10 500
        500 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages
        100 /var/lib/dpkg/status
     2.06-2ubuntu7 500
        500 http://fi.archive.ubuntu.com/ubuntu jammy/main amd64 Packages
```
and then filters out any packages that don't have a phase % or have a % other than 100.

As a disclaimer, I don't have a lot of exprience with tokio so there might be a cleaner way to express the `filter_map`/`Result`-mess.

### CODE REVIEW

File: src/blocks/apt.rs [hunk 1]
```diff
@@ -192,3 +198,54 @@ fn get_update_count(updates: &str) -> usize {
...
+                    Ok(true) => None,
+                    Ok(false) => Some(ResultBool(Ok(true))),
+                    Err(e) => Some(ResultBool(Err(e))),
+                }
+            })
+            .collect::<Vec<ResultBool>>()
+            .await
+            .into_iter()
+            .map(|ResultBool(v)| v)
+            .collect::<Result<Vec<bool>>>()?;
```
Comment (MaxVerevkin) [inline src/blocks/apt.rs:225]: I think this is a bit over-engineered. Writing this using a simple `for` loop is simpler, more straight-forward and avoids two unnecessary `.collect`s:

```rs
let mut cnt = 0;

for update_line in updates.lines().filter(|line| line.contains("[upgradable")) {
    if !is_phased_update(config_path, update_line).await? {
        cnt += 1;
    }
}

Ok(cnt)
```

Comment (sndels) [reply src/blocks/apt.rs]: Ah yes, derp. Good point.

File: src/blocks/apt.rs [hunk 2]
```diff
@@ -192,3 +198,38 @@ fn get_update_count(updates: &str) -> usize {
 fn has_matching_update(updates: &str, regex: &Regex) -> bool {
     updates.lines().any(|line| regex.is_match(line))
 }
+
+async fn get_update_count_ignore_phased(config_path: &str, updates: &str) -> Result<usize> {
+    let mut cnt = 0;
+
+    for update_line in updates.lines().filter(|line| line.contains("[upgradable")) {
+        if !is_phased_update(config_path, update_line).await? {
```
Comment (MaxVerevkin) [inline src/blocks/apt.rs:206]: You can also merge `get_update_count` and `get_update_count_ignore_phased` by doing something like

```rust
if !ignore_phased_updates || !is_phased_update(config_path, update_line).await? {
    cnt += 1;
}
```

Comment (sndels) [reply src/blocks/apt.rs:206]: Makes sense. I don't know why I expected weirdness with an async term in an or-chain, but looks like it does get skipped you'd expect :+1:

**Review by MaxVerevkin — APPROVED** [review]

**Comment by MaxVerevkin** [comment]
Thanks!

**Comment by sndels** [comment]
Thank you! Nice to have this integrated :+1:

**Pull request merged**